spec/sidekiq_poison_pill_remedy_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require "sidekiq"
+require "sidekiq/api"
+require "sidekiq/testing"
+require "rspec-sidekiq"
+require "sidekiq-poison-pill-remedy"
+require "support/my_job"
+
+RSpec.describe SidekiqPoisonPillRemedy do
+  before do
+    Sidekiq::Testing.inline!
+    allow(Sentry).to receive(:capture_message)
+  end
+
+  it "moves job to poison_pill queue and logs message" do
```

Comment:
to test this scenario, it would be enough  to have any kind of job that could be scheduled by `job = MyJob.peform_async(nil)` and verify side effects


```
job = MyJob.peform_async(nil)

# assert number of jobs in default queue
# assert number of jobs in poison_pill queue

SidekiqPoisonPillRemedy.remedy.call(job)

# assert number of jobs in default queue
# assert number of jobs in poison_pill queue

```
the side effect should be that in default queue the number of jobs will go from 1 to 0, and for poison pill from 0 to 1.

don't use inline mode, use a test one, to make sure the job will not be processed